sui_core/src/authority/authority_store.rs

```diff
@@ -209,6 +209,19 @@ impl<const ALL_OBJ_VER: bool, S: Eq + Serialize + for<'de> Deserialize<'de>>
         }
     }
 
+    /// Returns the TransactionEffects if we have a signed_effects structure for this transaction digest
+    pub fn get_effects(
+        &self,
+        transaction_digest: &TransactionDigest,
+    ) -> SuiResult<TransactionEffects> {
```

Comment:
I think this function can be more useful if it can return TransactionEffectsEnvelop